src/anm/analytics.js

```diff
@@ -70,6 +84,13 @@ Analytics.prototype.trackPlayer = function trackPlayer(name) {
     return function (player) { this.track(name, player); }.bind(this);
 };
 
+Analytics.prototype.sendPlayerData = function sendForPlayer(player) {
+    this.trackPlayingComplete(player);
+    var action = [this.queue[player.viewId]];
+    this.queue[player.viewId] = undefined;
+    this.sendData(action)
```

Comment:
Moving data from objects to arrays and back looks weird, actually, could this be changed to managing one data type? If you always send updates one by one, why do you need arrays? Here, anyway, you clear the hash cell for this `viewId`.